### PR TITLE
Internalizing types that were intended to only be used internally in IIS

### PR DESCRIPTION
**Pull request by jkotalik**
Missed types that were marked as public that should be marked internal.

### CODE REVIEW

**Comment by Tratcher** [comment]
Technically breaking, but probably not in practice. It was enough to fail API Check though...

**Comment by jkotalik** [comment]
Yeah this is "technically breaking", but it's effectively the same as a type in an internal namespace.

File: src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNetCore.Server.IIS.Core
     /// <summary>
     /// The default authentication handler with IIS In-Process
     /// </summary>
-    public class IISServerAuthenticationHandler : IAuthenticationHandler
+    internal class IISServerAuthenticationHandler : IAuthenticationHandler
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs:14]: Found a couple public projects that used this type, not sure how important it is to keep it, but just an FYI
https://github.com/halllo/IdentityServerExperiments/blob/267d6ba82a217d23e9fa966ad74505b717c42c28/IdentityServer/Startup.cs#L116
https://github.com/RegionOrebroLan/.NET-Web-Authentication-Extensions/blob/c96f94abc54df06ccaa8bf5f815225b3f95eb152/Source/Project/Extensions/TypeExtension.cs#L52

Comment (Tratcher) [reply src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs:14]: That first one seems like a very outdated workaround to set the DisplayName. I don't quite follow what that second one is doing.

**Comment by halter73** [comment]
Can we just throw ObsoleteAttributes on these and keep them public?

**Comment by jkotalik** [comment]
> Can we just throw ObsoleteAttributes on these and keep them public?

We could, but that isn't how we handled making all internal namespaces internal. Is this different because it isn't an internal namespace?

I'm not very concerned about breaking people with this change, however if we want to be extra careful, we can obsolete now and delete in a year.

**Comment by halter73** [comment]
When we first internalized pubternal types we decided it was worth the break for the massive reduction in public API surface. We could also at least say that the namespace contained ".Internal" so there was some warning.

I think obsoleting these types is okay if hardly anyone uses them and there's not valid use case. I don't see why we would ever delete them unless there was some reason we absolutely had to.

**Comment by jkotalik** [comment]
Any suggestions on what else to add besides mentioning they are obsolete? These types were never intended to be public, maybe just mentioning that is good enough?

**Comment by jkotalik** [comment]
@halter73 when we obsolete a type that we still want to use internally, is it typical that we create an internal pair to be used instead?

**Comment by halter73** [comment]
We should avoid breaking changes in obsolete APIs if possible just like we would for any other public API. If that means using a modified copy internally, that's fine. I don't think we need to make a copy preemptively though.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**